featureflags.go

```diff
@@ -193,28 +603,37 @@ func (poller *FeatureFlagsPoller) isSimpleFlagEnabled(key string, distinctId str
 
 // extracted as a regular func for testing purposes
 func checkIfSimpleFlagEnabled(key string, distinctId string, rolloutPercentage uint8) (bool, error) {
+	val, err := _hash(key, distinctId, "")
+
+	if err != nil {
+		return false, err
+	}
+
+	return val <= float64(rolloutPercentage)/100, nil
+}
+
+func _hash(key string, distinctId string, salt string) (float64, error) {
 	hash := sha1.New()
-	hash.Write([]byte("" + key + "." + distinctId + ""))
+	hash.Write([]byte("" + key + "." + distinctId + "" + salt))
 	digest := hash.Sum(nil)
 	hexString := fmt.Sprintf("%x\n", digest)[:15]
 
 	value, err := strconv.ParseInt(hexString, 16, 64)
 	if err != nil {
-		return false, err
+		return 0, err
 	}
-	return (float64(value) / LONG_SCALE) <= float64(rolloutPercentage)/100, nil
+
+	return float64(value) / LONG_SCALE, nil
+
 }
 
 func (poller *FeatureFlagsPoller) GetFeatureFlags() []FeatureFlag {
 	// ensure flags are loaded on the first call
+
 	if !poller.fetchedFlagsSuccessfullyOnce {
 		<-poller.loaded
 	}
 
-	poller.mutex.RLock()
-
-	defer poller.mutex.RUnlock()
-
 	return poller.featureFlags
 }
 
```

Comment:
Minor nit: This request method is very confusing (used for both decide and local_evaluation, based on things passed in.

I'd suggest extracting the common bits out, and having two fns, one which calls `/deicde`, and the other for local eval. (they do the auth setup / adding token to URL etc. etc.).

Also note: L691 -> Send group and person properties to decide!